pkg/hintrunner/zero/zerohint_math.go

```diff
@@ -501,3 +502,99 @@ func createSplitIntHinter(resolver hintReferenceResolver) (hinter.Hinter, error)
 	}
 	return newSplitIntHint(output, value, base, bound), nil
 }
+
+func newSplitFeltHint(maxHigh, maxLow, low, high, value hinter.ResOperander) hinter.Hinter {
+	return &GenericZeroHinter{
+		Name: "SplitFelt",
+		Op: func(vm *VM.VirtualMachine, _ *hinter.HintRunnerContext) error {
+			//> from starkware.cairo.common.math_utils import assert_integer\nassert ids.MAX_HIGH < 2**128 and ids.MAX_LOW < 2**128\nassert PRIME - 1 == ids.MAX_HIGH * 2**128 + ids.MAX_LOW\nassert_integer(ids.value)\nids.low = ids.value & ((1 << 128) - 1)\nids.high = ids.value >> 128
+
+			//> assert ids.MAX_HIGH < 2**128 and ids.MAX_LOW < 2**128
+			maxHigh, err := hinter.ResolveAsFelt(vm, maxHigh)
+			if err != nil {
+				return err
+			}
+			if !utils.FeltLt(maxHigh, &utils.FeltMax128) {
+				return fmt.Errorf("assertion `split_felt(): MAX_HIGH %v is out of range` failed", maxHigh)
+			}
+			maxLow, err := hinter.ResolveAsFelt(vm, maxLow)
+			if err != nil {
+				return err
+			}
+			if !utils.FeltLt(maxLow, &utils.FeltMax128) {
+				return fmt.Errorf("assertion `split_felt(): MAX_LOW %v is out of range` failed", maxLow)
+			}
+
+			//> assert PRIME - 1 == ids.MAX_HIGH * 2**128 + ids.MAX_LOW
+			leftHandSide := new(fp.Element).SetInt64(-1)
+			rightHandSide := new(fp.Element).Add(new(fp.Element).Mul(maxHigh, &utils.FeltMax128), maxLow)
+			if leftHandSide.Cmp(rightHandSide) != 0 {
+				return fmt.Errorf("assertion `split_felt(): The sum of MAX_HIGH and MAX_LOW does not equal to PRIME - 1` failed")
+			}
+
+			//> assert_integer(ids.value)
+			value, err := hinter.ResolveAsFelt(vm, value)
+			if err != nil {
+				return err
+			}
+
+			var valueBigInt big.Int
+			value.BigInt(&valueBigInt)
+			lowAddr, err := low.GetAddress(vm)
+			if err != nil {
+				return err
+			}
+
+			highAddr, err := high.GetAddress(vm)
+			if err != nil {
+				return err
+			}
+
+			//> ids.low = ids.value & ((1 << 128) - 1)
+			felt128 := new(big.Int).Lsh(big.NewInt(1), 128)
+			felt128 = new(big.Int).Sub(felt128, big.NewInt(1))
+			lowBigInt := new(big.Int).And(&valueBigInt, felt128)
+			lowValue := memory.MemoryValueFromFieldElement(new(fp.Element).SetBigInt(lowBigInt))
+
+			err = vm.Memory.WriteToAddress(&lowAddr, &lowValue)
+			if err != nil {
+				return err
+			}
+			//> ids.high = ids.value >> 128
+			highBigInt := new(big.Int).Rsh(&valueBigInt, 128)
+			highValue := memory.MemoryValueFromFieldElement(new(fp.Element).SetBigInt(highBigInt))
+
+			return vm.Memory.WriteToAddress(&highAddr, &highValue)
+
+		},
+	}
+}
+
+func createSplitFeltHinter(resolver hintReferenceResolver) (hinter.Hinter, error) {
+	maxHigh, err := resolver.GetResOperander("MAX_HIGH")
+	if err != nil {
+		return nil, err
+	}
+
+	maxLow, err := resolver.GetResOperander("MAX_LOW")
+	if err != nil {
+		return nil, err
+	}
```

Comment:
These two variables are constants, if you check the compiled output of a file that's using this hint, you'll see that they are not actually parameters you need to pass to the hint. The json looks like this:
```
   "15": [
            {
                "accessible_scopes": [
                    "starkware.cairo.common.math",
                    "starkware.cairo.common.math.split_felt"
                ],
                "code": "from starkware.cairo.common.math_utils import assert_integer\nassert ids.MAX_HIGH < 2**128 and ids.MAX_LOW < 2**128\nassert PRIME - 1 == ids.MAX_HIGH * 2**128 + ids.MAX_LOW\nassert_integer(ids.value)\nids.low = ids.value & ((1 << 128) - 1)\nids.high = ids.value >> 128",
                "flow_tracking_data": {
                    "ap_tracking": {
                        "group": 4,
                        "offset": 0
                    },
                    "reference_ids": {
                        "starkware.cairo.common.math.split_felt.high": 10,
                        "starkware.cairo.common.math.split_felt.low": 9,
                        "starkware.cairo.common.math.split_felt.range_check_ptr": 11,
                        "starkware.cairo.common.math.split_felt.value": 7
                    }
                }
            }
        ]
```
I would suggest you define these variables inside the hint implementation instead. Fixing this will require you to modify also the constructor of this hint and the test cases